erpnext-fhir-mapper/src/main/java/com/ozonehis/eip/mappers/erpnext/QuotationItemMapper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.mappers.erpnext;
+
+import com.ozonehis.eip.mappers.ToERPNextMapping;
+import com.ozonehis.eip.model.erpnext.QuotationItem;
+import org.hl7.fhir.r4.model.Dosage;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QuotationItemMapper<R extends Resource> implements ToERPNextMapping<R, QuotationItem> {
+
+    @Override
+    public QuotationItem toERPNext(R resource) {
+        QuotationItem quotationItem = new QuotationItem();
+        if (resource instanceof ServiceRequest serviceRequest) {
+            quotationItem.setCustomExternalID(serviceRequest.getIdPart());
+            if (serviceRequest.hasCode()) {
+                quotationItem.setItemCode(
+                        serviceRequest.getCode().getCodingFirstRep().getCode());
+            }
+            quotationItem.setQuantity(1); // default quantity is 1 for serviceRequests.
+            String requesterDisplay = serviceRequest.getRequester().getDisplay();
+            String serviceDisplay = serviceRequest.getCode().getText();
+            quotationItem.setDescription(serviceDisplay + " | Requester: " + requesterDisplay);
+
+        } else if (resource instanceof MedicationRequest medicationRequest) {
+            quotationItem.setCustomExternalID(medicationRequest.getIdPart());
+            Quantity quantity = medicationRequest.getDispenseRequest().getQuantity();
+            quotationItem.setQuantity(quantity.getValue().intValue());
+            quotationItem.setUnitOfMeasure(quantity.getUnit());
+            String medicationCode =
```

Comment:
So it is possible for the Reference here to be unspecified. Maybe we should make it so the bundle request downloads the actual `Medication` resource though, since that's more likely to have a useable / standardised code, whereas this is just going to be the UUID of the drug in the OpenMRS system, which isn't massively useful.

Reply:
In the `MedicationRequestProcessor` class, I'm already downloading the Medication resource as part of the bundle, I might as well utilize it to obtain the medication code/ID.

Reply:
Yeah, I think the codes would be better than just the UUID. Realistically, right now, we probably just have the UUID as the code anyway, but there's scope for future improvements.